examples/graphbolt/rgcn/hetero_rgcn.py

```diff
@@ -0,0 +1,638 @@
+(https://ogb.stanford.edu/docs/linkprop/)
+
+Paper [Modeling Relational Data with Graph Convolutional Networks]
+(https://arxiv.org/abs/1703.06103).
+
+Generation of graph embeddings is the main difference between homograph
+node classification and heterograph node classification:
+- Homograph: Since all nodes and edges are of the same type, embeddings
+  can be generated using a unified approach. Type-specific handling is
+  typically not required.
+- Heterograph: Due to the existence of multiple types of nodes and edges,
+  specific embeddings need to be generated for each type. This allows for
+  a more nuanced capture of the complex structure and semantic information
+  within the heterograph.
+
+This flowchart describes the main functional sequence of the provided example.
+main
+│
+├───> load_dataset
+│     │
+│     └───> Load dataset
+│
+├───> rel_graph_embed [HIGHLIGHT]
+│     │
+│     └───> Generate graph embeddings
+│
+├───> Instantiate RGCN model
+│     │
+│     ├───> RelGraphConvLayer (input to hidden)
+│     │
+│     └───> RelGraphConvLayer (hidden to output)
+│
+└───> run
+      │
+      │
+      └───> Training loop
+            │
+            ├───> EntityClassify.forward (RGCN model forward pass)
+            │
+            └───> validate and test
+                  │
+                  └───> EntityClassify.evaluate
+"""
+import argparse
+import itertools
+import sys
+
+import dgl.graphbolt as gb
+import dgl.nn as dglnn
+
+import psutil
+
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.nn import HeteroEmbedding
+from ogb.nodeproppred import Evaluator
+from tqdm import tqdm
+
+
+def load_dataset(dataset_name):
+    """Load the dataset and return the graph, features, train/valid/test sets
+    and the number of classes.
+    """
+    dataset = gb.BuiltinDataset(dataset_name).load()
+    print(f"Loaded dataset: {dataset.tasks[0].metadata['name']}")
+
+    graph = dataset.graph
+    features = dataset.feature
+    train_set = dataset.tasks[0].train_set
+    valid_set = dataset.tasks[0].validation_set
+    test_set = dataset.tasks[0].test_set
+    num_classes = dataset.tasks[0].metadata["num_classes"]
+
+    return graph, features, train_set, valid_set, test_set, num_classes
+
+
+def create_dataloader(
+    graph, features, item_set, device, batch_size, fanouts, shuffle, num_workers
+):
+    """Create a GraphBolt dataloader for training, validation or testing."""
+
+    # Initialize the ItemSampler to sample mini-batches from the dataset.
+    datapipe = gb.ItemSampler(item_set, batch_size=batch_size, shuffle=shuffle)
+
+    # Sample neighbors for each seed node in the mini-batch.
+    datapipe = datapipe.sample_neighbor(graph, fanouts=fanouts)
+
+    # Fetch the features for each node in the mini-batch.
+    datapipe = datapipe.fetch_feature(features, {"paper": ["feat", "year"]})
+
+    # Move the mini-batch to the appropriate device.
+    datapipe = datapipe.copy_to(device)
```

Comment:
Does this actually work?

Reply:
gpu not verified yet. But this is supposed to be canonical UX, so I leave it here.